### DISCUSSION TITLE
Eachine E110 heli support - chip: RF2520A (E129 = RF2418)

### DISCUSSION THREAD

**Discussion by vandusinae** [pascallanger/DIY-Multiprotocol-TX-Module#700] [General]
just checked the tx of the new Eachine E110 "stealth" heli and the chip (rf 2520A) looks very similar to the one in the E129 (chip rf2518) which MULTI supports. Here's a picture

![e110-2](https://user-images.githubusercontent.com/56637534/175781422-a5041edc-8dec-4d53-b790-c654129b9b75.jpg)

I've got only a cheap 8 channel logic analyzer and don't know much about protocols and nothing  about hacking protocols. But I may could record data with Saleae Logic, if someone tells me where to connect the cables and what to do.

**Comment by pascallanger** [comment]
There is nowhere to connect since it's an all in one chip with micro controller plus RF.
You can only reverse the protocol by listening over the air.
I'm currently adding the protocol of the C186 (eachine e200?) which might be the same as yours (using a rf2500 too). It will be available in the coming days. I can already fly the heli but I need to do some clean up and check some flags.

If the protocol is different then I would need the TX+heli to reverse it.

**Comment by vandusinae** [comment]
Thanks Pascal, I will test it.

Cheers

**Comment by vandusinae** [comment]
The radios are almost identical from the outside, only different stickers. The C186 looks like the "new" Eachine E120  https://www.youtube.com/watch?v=ntoiEowHb_Q

**Reply by pascallanger** [reply]
Yes e120 not e200...

**Reply by pascallanger** [reply]
The new builds having the E120/C186/K127 protocol are available here: https://downloads.multi-module.org/latest-test/
Documentation: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#e129---83
Please test and report.

**Comment by vandusinae** [comment]
The test build seems to work. Great work!

 https://youtu.be/MyA_SQnbPDw

**Comment by shafr** [comment]
Same chip is used for German's `Jamal Q4X` drone. (Sold this week in Lidl together with a helicopter for $20)

The bind process is - turn on Drone, turn on Remote, Left stick (Throttle) all the way UP and down. Then it binds.

However using `Jumper T-pro` and choosing `129` protocol and bind option i'm not able to bind.
Is there something else in the bind process ?

@vandusinae @pascallanger 

![image](https://user-images.githubusercontent.com/26896695/200123851-8bf3121e-a5bc-44f7-9eec-396c2c321459.png)

**Reply by pascallanger** [reply]
@shafr This is an all in one RF + ARM core so you can put whatever program you want in it and therefore whatever protocol... Having a RF25xx in it means that I should be able to reverse the protocol since I know partially how this chip works now.
I would need to get my hands on a TX and model to be able to reverse the protocol. I've checked and can't find any RC controlled models on the French Lidl web pages. Best would be to ship me a TX + model at my attention in France.

**Reply by shafr** [reply]
@pascallanger any update here ?

**Reply by pascallanger** [reply]
Not sure what you are waiting for.

**Reply by shafr** [reply]
Sorry, I sent a package with 3 devices (one of which is that drone, eachine drone, and airplane), as you mentioned above for analysis. (those are probably do not have specific model tags).

**Comment by Zoanie** [comment]
Perhaps this is just an aside. The same chip is used for a cheaply available ebay r/c stunt car I have been hacking. A google search turned up your thread. I appreciate your info on the device as I have found no other source. 

https://www.ebay.com/itm/354406168441
![IMG_6581](https://user-images.githubusercontent.com/5397340/226075730-9466f934-da5f-4944-a5b1-134e05d0bafa.jpg)
![IMG_6580](https://user-images.githubusercontent.com/5397340/226075738-28b08c7b-4dcb-4d5a-b9cf-0f767d7bd53e.jpg)

    Cheers,

Carlos

**Comment by Zoanie** [comment]
I'm just providing a comment on usage of the RF2520A. If you do want to
reverse engineer the operations. The link provides an inexpensive option.

Carlos

**Comment by Silviu-Marian** [comment]
In case it helps, chip is MIXIC	RF2520A which under the hood is a Texas Instruments CC2520.

The same chip should be in the remote where it is programmed for remote control, and in a drone/car/boat etc. where it controls the motors, combines sensor data, transmits AV etc. In some toys it can also create a wi-fi network and comes with a phone app which can receive video and act as a remote.

There's ISCP available, might be worth trying debug mode with it. If not, I've attached CC2520 programming instructions for RF2520B.

Datasheet:
[RF2520A_B-83413cfcbe276235867c2a72a7850bb5.pdf](https://github.com/user-attachments/files/16097482/RF2520A_B-83413cfcbe276235867c2a72a7850bb5.pdf)

CC2520 programming:
[CC2520-20100729091640613.pdf](https://github.com/user-attachments/files/16097489/CC2520-20100729091640613.pdf)

Source(s):
[2.4Ghz Zigbee wireless module](http://icshopping.com.tw/product-page.php?5270)
<img width="448" alt="CC2520-RF2520B" src="https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/1747499/cb20c5cd-6d68-4b69-b9cf-c49f43f6a622">
